### DISCUSSION TITLE
Include list of ADRs

### DISCUSSION THREAD

**Discussion by eNBeWe** [docToolchain/docToolchain#563] [Q&A] [answered]
Hi,

we are starting to user doc-as-code and especially docToolchain right now.
Today I started writing the first architecture decision record and wanted to include that in the documentation, leading to this problem:

Is there a way to have all files from a subdirectory gatered up in the doc?
I thought it would be a good idea to have one file per ADR and put them in a subdir of the `src` directory.
In the `09_design_decisions.adoc` I then would like to have a list (like a TOC) with all the ADRs.

Any ideas how I could do that? Or am I doing it just plain wrong and should do something differently?

**Comment by uniqueck** [comment]
Hi @eNBeWe,

in general there is no way for that, but you can add every new adr via include macro to the parent 09_design_decision.adoc page. 
An other approach is to write a custom include macro extension for that. Check this example https://docs.asciidoctor.org/asciidoctorj/latest/extensions/include-processor/, but beware of asciidoctorj version in doctoolchain, you have to use an older version, to be compatible with doctoolchain.

**Comment by uniqueck** [comment, marked as answer]
Oh sorry I'm wrong. In doctoolchain exists a script to include a set of files to a parent document. See https://github.com/docToolchain/docToolchain/blob/master/scripts/collectIncludes.gradle

**Comment by eNBeWe** [comment]
That looks like exactly what I was looking for. Thanks a lot! Will look into that.
